proj/src/main/webapp/index.html

```diff
@@ -20,8 +20,11 @@
         <div id="content"></div>
         <ul id="chat-as-list"></ul>
         <form>
+            <!-- this button stops form submission on pressing enter
+            which is not what we want because it reloads the page -->
+            <button type="submit" disabled style="display: none" aria-hidden="true"></button>
```

Comment:
I don't think a hack like this is necessary to prevent form submission.  Try adding `event.preventDefault()` to the key listener in your JavaScript. <https://stackoverflow.com/questions/19103350/dont-submit-form-when-i-press-enter-key-in-a-textbox>  I also think that there's no particular reason to keep using an HTML form here, if you want to fix the problem by getting rid of that instead.

Reply:
so the preventDefault function didn't work for some reason, so I ended up just removing the form element

Reply:
Does the input text and button need. to be inside a form? It seems like the form tag might not be adding anything useful besides maybe some built in styles.

Reply:
yeah I was doing it so it could be more separate in something more specific than a div element, but it seems like i'll just have to leave it out